src/Build/Graph/ProjectGraph.cs

```diff
@@ -18,12 +22,22 @@ public sealed class ProjectGraph
     {
         private const string ProjectReferenceItemName = "ProjectReference";
         private const string FullPathMetadataName = "FullPath";
+        private const string ToolsVersionMetadataName = "ToolsVersion";
+        private const string PropertiesMetadataName = "Properties";
+        private const string UndefinePropertiesMetadataName = "UndefineProperties";
+        private const string AdditionalPropertiesMetadataName = "AdditionalProperties";
+        private const string SetConfigurationMetadataName = "SetConfiguration";
+        private const string SetPlatformMetadataName = "SetPlatform";
+        private const string SetTargetFrameworkMetadataName = "SetTargetFramework";
+        private const string GlobalPropertiesToRemoveMetadataName = "GlobalPropertiesToRemove";
         private const string ProjectReferenceTargetsItemType = "ProjectReferenceTargets";
         private const string ProjectReferenceTargetsMetadataName = "Targets";
         private const string DefaultTargetsMarker = ".default";
 
-        private readonly Dictionary<string, ProjectGraphNode> _allParsedProjects =
-            new Dictionary<string, ProjectGraphNode>(StringComparer.OrdinalIgnoreCase);
+        private static readonly char[] PropertySeparator = { ';' };
+
+        private readonly Dictionary<ConfigurationMetadata, ProjectGraphNode> _allParsedProjects =
```

Comment:
>_allParsedProjects [](start = 77, length = 18)

I don't understand the significance of having ConfigurationMetadata as the the key? Do we expect to have multiple entries for the same project path with different tools version and properties?

Reply:
Yes, that's exactly it. Different projects may depend on a project with different sets of global properties. This causes that project to build multiple times with those different sets of properties.